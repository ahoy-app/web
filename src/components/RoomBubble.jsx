import React from 'react'
import PropTypes from 'prop-types'
import { Link } from 'react-router-dom'

import { color } from '../style'
import T from './Text'

const styles = {
  background: color.background.light,
  height: '40px',
  width: '40px',

  borderRadius: '50%',

  textAlign: 'center',
  lineHeight: '40px',
}

const RoomBubble = ({ room }) => (
  <div style={styles}>
    <Link to={`/${room.id}`} style={{ textDecoration: 'none' }}>
      <T.title_2>{room.name.charAt(0).toUpperCase()}</T.title_2>
    </Link>
  </div>
)

RoomBubble.propTypes = {
  room: PropTypes.object.isRequired,
}

export default RoomBubble
