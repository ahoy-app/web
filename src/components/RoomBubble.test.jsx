import React from 'react'
import { shallow } from 'enzyme'

import RoomBubble from './RoomBubble'

const room = 'mclarens'

describe('RoomBubble', () => {
  it('should render correctly', () => {
    const component = shallow(<RoomBubble room={room} />)
    expect(component).toMatchSnapshot()
  })
})
