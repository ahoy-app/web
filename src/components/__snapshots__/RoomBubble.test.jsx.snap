// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoomBubble should render correctly 1`] = `
<div
  style={
    Object {
      "background": "#fff",
      "borderRadius": "50%",
      "height": "40px",
      "lineHeight": "40px",
      "textAlign": "center",
      "width": "40px",
    }
  }
>
  <Link
    style={
      Object {
        "textDecoration": "none",
      }
    }
    to="/mclarens"
  >
    <Text-title_2>
      M
    </Text-title_2>
  </Link>
</div>
`;
